src/components/AddressBook/SetContactDialog.vue

```diff
@@ -0,0 +1,251 @@
+<template>
+  <dialog-base :title="title" :visible.sync="isVisible" :tooltip="tooltip">
+    <div class="set-address">
+      <s-input
+        ref="address"
+        class="set-address__input"
+        :placeholder="t('nameText')"
+        v-model="name"
+        :disabled="loading"
+        :maxlength="30"
+      />
+      <s-input class="set-address__input" :placeholder="t('addressText')" v-model="address" :disabled="inputDisabled" />
+      <p v-if="isAccountAddress" class="set-address-error">{{ t('walletSend.addressError') }}</p>
+      <template v-if="validAddress && isNotSoraAddress">
+        <p class="wallet-send-address-warning">{{ t('addressBook.notSoraAddress') }}</p>
+        <s-tooltip :content="copyValueAssetId" placement="top">
+          <p class="wallet-send-address-formatted" @click="handleCopyAddress(formattedSoraAddress, $event)">
+            {{ formattedSoraAddress }}
+          </p>
+        </s-tooltip>
+      </template>
+      <s-input
+        class="set-address__input"
+        :placeholder="t('addressBook.identity')"
+        v-model="onChainIdentity"
+        :disabled="true"
+      />
+      <div class="set-address__btn">
+        <s-button @click="setContact" type="primary" class="s-typography-button--large" :disabled="btnDisabled">
+          {{ btnText }}
+        </s-button>
+      </div>
+    </div>
+  </dialog-base>
+</template>
+
+<script lang="ts">
+import debounce from 'lodash/debounce';
```

Comment:
`lodash/fp/debounce` -> to reduce number of exported files (check rollup config please)